src/Transports/MassTransit.AmazonSqsTransport/Pipeline/AmazonSqsMessageReceiver.cs

```diff
@@ -170,5 +241,39 @@ async Task ActiveAndActualAgentsCompleted(StopContext context)
                 }
             }
         }
+
+        class SequenceNumberComparer : IComparer<string>
+        {
+            public static readonly SequenceNumberComparer Instance = new SequenceNumberComparer();
+
+            public int Compare(string x, string y)
+            {
+                if (string.IsNullOrEmpty(x))

```

Comment:
what would be if both `x` and `y` are null or empty?

Reply:
exception

Reply:
I mean, is that expected? why do we wanna throw, I mean not all messages might be ordered considering backward compatibility, etc

Reply:
It's not about a backward compatibility. All messages in FIFO queue must have SequenceNumber. It's impossible case when message doesn't have  SequnceNumber.

Reply:
okay, so this responsibility is not part of comparator then. It should be in receiver